### ISSUE TITLE
Add a new rule to replace stateful class names with appropriate native or ARIA attributes

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#1759]
## Overview
When a `button` element is assigned a class like `xxx--on`, the state should typically be managed using attributes such as `aria-pressed`. Rather than relying solely on visual state changes through class and style, it's crucial to expose these state changes to assistive technologies.

## Proposal
Create a new rule that detects class names indicating states like `on`/`off` or `opened`/`closed`, and suggest replacing them with native or ARIA attributes. This rule aims to achieve the following improvements:
- Enhanced accessibility
- Improved code consistency and readability
- Compliance with HTML standards and WAI-ARIA specifications

## Tasks
1. Add logic to detect class names such as `xxx--on`, `xxx--off`, `xxx--opened`, and `xxx--closed`.
2. Display warning messages suggesting appropriate native attributes (e.g., `aria-pressed`, `aria-expanded`) when such class names are detected.
3. Update user documentation to include an explanation of this new rule.

## Example Warning Messages

"Use `aria-pressed` instead of `xxx--on` class to manage the pressed state."

Adding this new rule will enhance accessibility and code quality, ensuring that state changes are properly exposed to users of assistive technologies.

**Comment by YusukeHirao** [comment]
This can already be achieved with the existing configuration using `nodeRules` + attribute selectors + the `required-attr` rule. No new rule is needed.

### Example Configuration

```json
{
  "nodeRules": [
    {
      "selector": "button[class*='--on'], button[class*='--off']",
      "rules": {
        "required-attr": "aria-pressed"
      }
    },
    {
      "selector": "[class*='--opened'], [class*='--closed']",
      "rules": {
        "required-attr": "aria-expanded"
      }
    }
  ]
}
```

### How It Works

- **`[class*="--on"]`** — CSS attribute selector that matches elements whose `class` contains `--on` (partial match)
- **`required-attr`** — Rule that requires specific attributes to be present
- **`nodeRules`** — Allows you to apply rule configurations to elements matching a specific selector

You can customize the selectors and required attributes to fit your project's naming conventions. For more details, see:
- [nodeRules documentation](https://markuplint.dev/docs/configuration/properties#noderules)
- [required-attr rule documentation](https://markuplint.dev/docs/rules/required-attr)

Closing this as it's achievable with existing features. Feel free to reopen if you have any questions.
